internal/monitor/monitor.go

```diff
@@ -245,27 +245,51 @@ func (pm *PowerMonitor) isFresh() bool {
 	return age <= pm.maxStaleness
 }
 
-// refreshSnapshot create a new snapshot of the power consumption of various levels( currently only node)
+// refreshSnapshot creates a new snapshot of the power consumption
+// It handles both initial and subsequent collections assuming previous Snapshot
+// is nil only on first call.
 func (pm *PowerMonitor) refreshSnapshot() error {
 	started := pm.clock.Now()
-	defer func() { pm.logger.Info("Computed power", "duration", pm.clock.Since(started)) }()
+	defer func() {
+		pm.logger.Info("Computed power", "duration", pm.clock.Since(started))
+	}()
 
+	newSnapshot := NewSnapshot()
 	prevSnapshot := pm.snapshot.Load()
-	// ensure snapshot is not nil from here on
-	if prevSnapshot == nil {
-		prevSnapshot = NewSnapshot()
-	}
 
-	newSnapshot := NewSnapshot()
-	if err := pm.calculateNodePower(newSnapshot.Node, prevSnapshot.Node); err != nil {
-		return fmt.Errorf("failed to calculate node power %w", err)
+	if prevSnapshot == nil {
+		// Handle initial collection explicitly
+		if err := pm.firstReading(newSnapshot); err != nil {
+			return err
+		}
+	} else {
+		if err := pm.calculatePower(prevSnapshot, newSnapshot); err != nil {
+			return err
+		}
 	}
 
-	// update snapshot
+	// Update snapshot with current timestamp
 	newSnapshot.Timestamp = pm.clock.Now()
 	pm.snapshot.Store(newSnapshot)
-
 	pm.signalNewData()
 
 	return nil
 }
+
+const (
+	nodePowerError = "failed to calculate node power: %w"
+)
+
+func (pm *PowerMonitor) firstReading(newSnapshot *Snapshot) error {
+	if err := pm.firstNodeRead(newSnapshot.Node); err != nil {
+		return fmt.Errorf(nodePowerError, err)
+	}
+	return nil
+}
+
+func (pm *PowerMonitor) calculatePower(prev, newSnapshot *Snapshot) error {
+	if err := pm.calculateNodePower(prev.Node, newSnapshot.Node); err != nil {
+		return fmt.Errorf(nodePowerError, err)
+	}
+	return nil
+}
```

Comment:
nit: these methods are called nowhere else, not even in test code, so why not inline these methods at the place these are called?

 not a blocker.

Reply:
This will expand in future to ... 

```
func firstReading(snapshot) {
  firstNodeRead(snapshot.Node)
  firstProcessRead(snapshot.Process)
  firstContainerRead(snapshot.Container)
  firstVMRead(snapshot.VM)
}

func computePower(prev, new) {
  computeNodePower(prev.Node, new.Node)
  computeProcessRead(prev.Processes, new.Processes)
  // .. container
  // .. VM
  // .. Pods 
}

```